### DISCUSSION TITLE
Dax Studio data Export to Amazon Redshift

### DISCUSSION THREAD

**Discussion by dalerjon-us** [DaxStudio/DaxStudio#786] [Q&A]
Hello everyone. 

1. The data comes via an api connection to DaxStudio and I want to export it to Amazon Redshift server. I am trying to export data from DaxStudio to an external server(Panoply, which uses Amazon Redshift). Is it possible to achieve this task or has anyone done this before. Currently, I am only able to export via CSVs and SQL Server(don't have). 

2. Is there a way to automate the exporting process given a specific time window in order to eliminate manual process?

**Comment by dgosbell** [comment]
DAX Studio does not store any data so I'm guessing that you are probably pulling it into a Power BI Dataset and then connecting to that using DAX Studio. For the SQL Server export we use the bulk copy API to do high speed, low memory usage exports, but this API is specific to SQL Server, we would need to build a different mechanism to export to other databases. It might be possible to build something more generic that could write out to via OLEDB or ODBC, but it would be a lot more work and it's not something we have on our short term road map. Outside of DAX Studio you could maybe look at using something like Azure Data Factory or something in Amazon like Lambdas to pull directly from the API into Redshift.

> 2. Is there a way to automate the exporting process given a specific time window in order to eliminate manual process? 

No, there is no way of automating this at the moment, but I hope to add a command line interface later this year which would allow for certain operations like exports to be automated.
